packages/server/src/synchronizer/reporter.ts

```diff
@@ -0,0 +1,51 @@
+import {through} from './streams'
+import File from 'vinyl'
+import {success, clearLine} from '../log'
+
+export type Event<T> = {type: string; payload: T}
+
+export const INIT = 'INIT'
+export const FILE_WRITTEN = 'FILE_WRITTEN'
+export const ERROR_THROWN = 'ERROR_THROWN'
+export const READY = 'READY'
+
+/**
+ * Reporter is a stream that converts build status events and prepares them for the console.
+ * A good way to think about this is as the root of the "view" component of the application.
+ */
+export default function createReporter() {
+  let lastEvent: Event<any> = {type: INIT, payload: null}
+
+  const stream = through({objectMode: true}, (event: Event<File>, _, next) => {
+    switch (event.type) {
+      case FILE_WRITTEN: {
+        const filePath = event.payload.history[0].replace(process.cwd(), '')
+        clearLine(filePath)
+        break
+      }
+
+      case ERROR_THROWN: {
+        // Tidy up if operational error is encountered
+        if (lastEvent.type === FILE_WRITTEN) {
+          clearLine()
+        }
+        break
+      }
+
+      case READY: {
+        if (lastEvent.type === FILE_WRITTEN) {
+          clearLine()
+        }
+        success('Blitz is ready')
+        break
+      }
+    }
+
+    // Capture last event incase we need to tidy up the console
+    lastEvent = event
+
+    next()
+  })
+
+  return {stream}
+}
```

Comment:
@merelinguist Might be good if you can check this out too

Reply:
If you squint it is sort of working like Redux...

Reply:
Hm do the events just come from us, or from Next/elsewhere?

Reply:
They come from us in the rules. You can search for the event type to find the source. I might have a play after this is merged with trying to get some oversight into next